voting_metadata.go

```diff
@@ -0,0 +1,52 @@
+package pbft
+
+// NewVotingMetadata initializes instance of VotingMetadata
+func NewVotingMetadata(votingPower map[NodeID]uint64) *VotingMetadata {
+	v := &VotingMetadata{votingPower: votingPower}
+	v.maxFaultyVotingPower = v.calculateMaxFaultyVotingPower()
+	v.quorumSize = 2*v.MaxFaultyVotingPower() + 1
+	return v
+}
+
+// VotingMetadata holds weighted vote for each validator. Weighted vote is based on its voting power (e.g. stake amount).
+// Voting power is used as input to calculate:
+//  - QuorumSize of entire validator set, M
+//  - MaxFaultyVotingPower, which is maximum tollerable value of malicious voting power in order to have Byzantine fault tollerant property satisfied.
+type VotingMetadata struct {
```

Comment:
You do not need this `VotingMetadata` anymore, if `ValidatorSet` returns the voting power, add two atomic functions that return the Quorum and MaxFaultyVotingPower whenever the sequence starts and move CalculateVotingPower to the state.

Reply:
The main purpose of `VotingMetadata` is that it can be reused in v3 client (we are using `QuorumSize` and `CalculateVotingPower`). If I'd move all of these to `state.go`, then we wouldn't be able to reuse it on v3 codebase. Agreed?

Reply:
But you can expose a function instead of a data structure
```
func CalculateQuorumPower() (uint64, uint64) {
   ...
}
```
And `CalculateVotingPower` is just an aggregation over a map. There is no point in having v3 depend on an external function to do that.

Reply:
Not only that, but also using `QuorumSize` as well: https://github.com/maticnetwork/v3/commit/656b17e00f8bfc68d29ac3f35f6f5c748a72aa05#diff-297043a61d1a54ac38dc24c4a73a7de2fac85bd5567b58b1cfb47d8bdff663bdR369.
I know that these functions are pretty simple and straight-forward to implement, but not sure why one would have same code on multiple places, since it is easy to reuse it via this small and encapsulated struct?

However, don't want to debate around this anymore, think that we've lost too much time discussing some relatively non-important points and decisions made. Will try to accommodate this feedback as well tomorrow.